src/verifier/tests.rs

```diff
@@ -224,3 +224,69 @@ pub fn test_type_errors() {
         assert!(!&res);
     }
 }
+#[test]
+pub fn test_user_defined_errors() {
+    setup_logger();
+    let mut err = ErrorGen::new("".to_string(), "".to_string(), 0);
+
+    let script = r#"
+        i32 a;
+        nested_fn() -> bool {
+            return "hi";
+            return 1;
+        }
+        dummy_fn() {
+            a = nested_fn();
+        }
+        wasm::call:alt {
+            dummy_fn();
+        }   
+    "#;
+    info!("Typechecking: {}", script);
+    let res = is_valid_script(script, &mut err);
+
+    err.report();
+    assert!(err.has_errors);
+    assert!(!res);
+}
+#[test]
+pub fn test_user_defined() {
+    setup_logger();
+    let mut err = ErrorGen::new("".to_string(), "".to_string(), 0);
+
+    let script = r#"
+        i32 a;
+        nested_fn() -> i32 {
+            return 5;
+        }
+        dummy_fn() {
+            a = nested_fn();
+        }
+        wasm::call:alt {
+            dummy_fn();
+        }   
+    "#;
+    info!("Typechecking: {}", script);
+    let res = is_valid_script(script, &mut err);
+
+    err.report();
+    assert!(!err.has_errors);
+    assert!(res);
+}
+// #[test]
```

Comment:
put a comment here that points to `build_visitor` saying we don't support `BEGIN`